var/spack/repos/builtin/packages/py-jsonnet/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class PyJsonnet(PythonPackage):
+    """A data templating language for app and tool developers based on JSON"""
+
+    homepage = "https://jsonnet.org/"
+    git = "https://github.com/google/jsonnet.git"
+
+    maintainers = ["jcpunk"]
```

Comment:
Since it's the same repository, can we get by with a single package that conditionally extends Python?

Reply:
The jsonnet python bits use a different build mechanism than the C bits.  While they come out of the same repo, they build in very different ways.

Is there an example package I can look at that will let me build the C bits via `make` and the python bits via `setuptools`?

Reply:
Not that I know of. @adamjstewart What would you prefer in this case? A unique package of two packages pointing at the same repository?

Reply:
This is a tough question, I think it depends on how the build system works. If it's possible to build the C library a single time (`jsonnet`) and then have the Python bindings depend on that installation (`py-jsonnet`) then I think it makes sense to have separate packages. If it isn't possible to build the Python bindings without rebuilding the C library, then I think we can have a single package.

Reply:
Currently the python bindings statically link the C library into the python object.  The build tools provided by google are doing this on purpose.  Each target (C or python) can be built and run independently with different versions installed.

Without a fair bit of patching the upstream sources, it isn't currently possible to build the python object without rebuilding the C code.

Reply:
I guess I'm fine with either having separate packages or having a single package with a `+python` variant. If you want to do the latter, grep the `package.py` files for packages that call `pip(...)`.

Reply:
I'll try to use that to merge into one unit.